source/interface.cc

```diff
@@ -0,0 +1,449 @@
+{
+  const unsigned dofs_per_cell = data.local_dof_indices.size();
+  cell->get_dof_indices (data.local_dof_indices);
+
+  data.local_matrix = 0;
+
+  get_preconditioner_residual(cell, scratch, data, data.sacado_residual);
+
+  for (unsigned int i=0; i<dofs_per_cell; ++i)
+    for (unsigned int j=0; j<dofs_per_cell; ++j)
+      data.local_matrix(i,j) = data.sacado_residual[i].dx(j);
+
+}
+
+
+template <int dim, int spacedim, int n_components, typename LAC>
+shared_ptr<Mapping<dim,spacedim> >
+Interface<dim,spacedim,n_components,LAC>::get_mapping(const DoFHandler<dim,spacedim> &,
+                                                      const typename LAC::VectorType &) const
+{
+  return shared_ptr<Mapping<dim,spacedim> >(new MappingQ<dim,spacedim>(1));
+}
+
+template <int dim, int spacedim, int n_components, typename LAC>
+UpdateFlags
+Interface<dim,spacedim,n_components,LAC>::get_jacobian_flags() const
+{
+  return (update_quadrature_points |
+          update_JxW_values |
+          update_values |
+          update_gradients);
+}
+
+template <int dim, int spacedim, int n_components, typename LAC>
+UpdateFlags
+Interface<dim,spacedim,n_components,LAC>::get_residual_flags() const
+{
+  return get_jacobian_flags();
+}
+
+template <int dim, int spacedim, int n_components, typename LAC>
+UpdateFlags
+Interface<dim,spacedim,n_components,LAC>::get_jacobian_preconditioner_flags() const
+{
+  return (update_JxW_values |
+          update_values |
+          update_gradients);
+}
+
+template <int dim, int spacedim, int n_components, typename LAC>
+UpdateFlags
+Interface<dim,spacedim,n_components,LAC>::get_face_flags() const
+{
+  return get_jacobian_flags();
+}
+
+template <int dim, int spacedim, int n_components, typename LAC>
+void
+Interface<dim,spacedim,n_components,LAC>::fix_solution_dot_derivative(FEValuesCache<dim,spacedim> &, double) const
+{}
+
+template <int dim, int spacedim, int n_components, typename LAC>
+void
+Interface<dim,spacedim,n_components,LAC>::fix_solution_dot_derivative(FEValuesCache<dim,spacedim> &fe_cache, Sdouble alpha) const
+{
+  auto &sol = fe_cache.get_current_independent_local_dofs("solution", alpha);
+  auto &sol_dot = fe_cache.get_current_independent_local_dofs("solution_dot", alpha);
+
+  for (unsigned int i=0; i<sol.size(); ++i)
+    sol_dot[i] = alpha.val()*sol[i] + (sol_dot[i].val() - alpha.val()*sol[i].val());
+}
+
+template <int dim, int spacedim, int n_components, typename LAC>
+void
+Interface<dim,spacedim,n_components,LAC>::fix_solution_dot_derivative(FEValuesCache<dim,spacedim> &fe_cache, SSdouble alpha) const
+{
+  auto &sol = fe_cache.get_current_independent_local_dofs("solution", alpha);
+  auto &sol_dot = fe_cache.get_current_independent_local_dofs("solution_dot", alpha);
+
+  for (unsigned int i=0; i<sol.size(); ++i)
+    sol_dot[i] = (alpha.val().val()*sol[i]) + (sol_dot[i].val().val() - alpha.val().val()*sol[i].val().val());
+}
+
+
+template <int dim, int spacedim, int n_components, typename LAC>
```

Comment:
I have some problem with the explicit instantiation of this method. @asartori86 and @luca-heltai any suggestion?